src/app/handlers/acknowledge.handler.ts

```diff
@@ -0,0 +1,37 @@
+import { GuildChannel, MessageReaction, User, Role } from 'discord.js';
+import { IContainer, IHandler } from '../../common/types';
+import Constants from '../../common/constants';
+import { Maybe } from '../../common/types';
+
+export class AcknowledgeHandler implements IHandler {
+  private _CoC_Channel: Maybe<GuildChannel> = undefined;
+  private _targetRole: Maybe<Role> = undefined;
+
+  constructor(public container: IContainer) {}
+
+  public async execute(reaction: MessageReaction, user: User): Promise<void> {
+    const member = this.container.guildService.get().members.get(user.id);
+    if (!member) {
+      return;
+    }
+
+    if (!this._CoC_Channel) {
+      this._CoC_Channel = member.guild.channels.find(
+        (c) => c.name === Constants.Channels.Public.CodeOfConduct
+      );
+    }
+
+    if (reaction.message.channel !== this._CoC_Channel) {
+      return;
+    }
+
+    if (!this._targetRole) {
+      this._targetRole = member.guild.roles.filter((r) => r.name === 'Un Acknowledged').first();
```

Comment:
i thought you had changed 'Un Acknowledged' to a class constant?

Reply:
The `Maybe<Role>` is a class property, you want me to move the `name:string` up there too?

Reply:
That was a class constant in new_member.handler